src/content/hinting.ts

```diff
@@ -87,6 +96,142 @@ class HintState {
         this.focusedHint = activeHints[nextFocusedIndex]
         this.focusedHint.focused = true
     }
+
+    changeFocusedHintTop() {
```

Comment:
There's a reasonable amount of redundancy in these changeFocusedHint* functions. If you fancied refactoring it so that the filter and reduction could be reused, I'd merge that PR :)